cpp/inline/games/stochastic_nim/Game.inl

```diff
@@ -0,0 +1,97 @@
+#include <games/stochastic_nim/Game.hpp>
+
+namespace stochastic_nim {
+
+inline void Game::Rules::init_state(State& state) {
+  state.stones_left = kStartingStones;
+  state.current_player = 0;
```

Comment:
You need to initialize `current_mode` here.

I think with this bug, when using `GameServer` to run G games with parallelism p with G > p, if a game ends in chance-mode, the next game launched on that game-thread will start in chance-mode.